panda/src/collide/collisionBox.cxx

```diff
@@ -493,6 +494,102 @@ test_intersection_from_ray(const CollisionEntry &entry) const {
   return new_entry;
 }
 
+PT(CollisionEntry) CollisionBox::
+test_intersection_from_parabola(const CollisionEntry &entry) const {
+  const CollisionParabola *parabola;
+  DCAST_INTO_R(parabola, entry.get_from(), nullptr);
+
+  const LMatrix4 &wrt_mat = entry.get_wrt_mat();
+
+  // Convert the parabola into local coordinate space.
+  LParabola local_p(parabola->get_parabola());
+  local_p.xform(wrt_mat);
+
+  PN_stdfloat t = INT_MAX;
+  PN_stdfloat cur_t;
+  PN_stdfloat t1, t2;
+  int intersecting_face = -1;
+  for (int i = 0; i < get_num_planes(); i++) {
+    LPlane face = get_plane(i);
+    bool intersects = false;
+
+    if (face.intersects_parabola(t1, t2, local_p)) {
+      // t1 and t2 are both candidates for intersection
+      // if they are not less than t, no need to check them
+      // otherwise, check if they are in our parabola
+      if (t1 < t && t1 >= parabola->get_t1() && t1 <= parabola->get_t2()){
+        intersects = true;
+        cur_t = t1;
+      }
+
+      if (t2 < t && t2 >= parabola->get_t1() && t2 <= parabola->get_t2()){
+        if (intersects) {
+          // both t1 and t2 intersect, take the smaller of the two.
+          cur_t = std::min(t1, t2);
+        } else {
+          cur_t = t2;
+          intersects = true;
+        }
+      }
+
+      if (!intersects) {
+        // the parabola does not intersect this face, skip to next one.
+        continue;
+      }
+
+      bool behind = true;
+      // we now check if the intersection point
+      // is behind all of the other planes.
+      for (int j = 0; j < get_num_planes(); j++) {
+        if (j == i) {
+          // no need to check the intersecting face
+          continue;
+        }
+        if (get_plane(j).dist_to_plane(local_p.calc_point(cur_t)) > 0.0f) {
+          // our point is in front of this face, so turns out
+          // it does not collide with the box at this point
+          behind = false;
+          break;
+        }
+      }
+
+      if (behind) {
```

Comment:
This probably does not solve your current problem, but shouldn't this check if `cur_t` is smaller than `t` so we get the first `intersecting_face` and `t` value?

Reply:
The check is actually done beforehand on lines 520 and 525:
https://github.com/panda3d/panda3d/pull/664/files#diff-6105c95465dfdba16cebccbd91fdd51eR520

Reply:
Ah up, I see it now.